From 6e68469f7f40fb10f1e47e167968f81e81b595e6 Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Sun, 2 Jan 2011 10:49:36 -0600
Subject: [PATCH 105/122] core: handle D-Bus disconnect better on exit

We're just going to quit here anyway, so we don't need to
make sure the property filter is unregistered.  Fixes a
crash on exit if D-Bus has quit before NM has.
---
 src/nm-manager.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/nm-manager.c b/src/nm-manager.c
index eb393fa..c7abf2d 100644
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -4351,10 +4351,11 @@ dispose (GObject *object)
 
 	/* Unregister property filter */
 	bus = nm_dbus_manager_get_connection (priv->dbus_mgr);
-	g_assert (bus);
-	dbus_connection = dbus_g_connection_get_connection (bus);
-	g_assert (dbus_connection);
-	dbus_connection_remove_filter (dbus_connection, prop_filter, manager);
+	if (bus) {
+		dbus_connection = dbus_g_connection_get_connection (bus);
+		g_assert (dbus_connection);
+		dbus_connection_remove_filter (dbus_connection, prop_filter, manager);
+	}
 	g_object_unref (priv->dbus_mgr);
 
 	if (priv->bluez_mgr)
-- 
1.7.3.4

